rust/agama-software/src/service.rs

```diff
@@ -202,57 +290,25 @@ impl MessageHandler<message::GetConfig> for Service {
 #[async_trait]
 impl MessageHandler<message::SetConfig<Config>> for Service {
     async fn handle(&mut self, message: message::SetConfig<Config>) -> Result<(), Error> {
-        let product = message.product.read().await;
+        self.product = Some(message.product.clone());
 
-        let software = {
+        {
             let mut state = self.state.write().await;
             state.config = message.config.clone().unwrap_or_default();
-            SoftwareState::build_from(&product, &state.config, &state.system, &self.selection)
-        };
+        }
 
         self.events.send(Event::ConfigChanged {
             scope: Scope::Software,
         })?;
 
-        tracing::info!("Wanted software state: {software:?}");
-
-        let model = self.model.clone();
-        let issues = self.issues.clone();
-        let events = self.events.clone();
-        let progress = self.progress.clone();
-        let product_spec = product.clone();
-        let state = self.state.clone();
-        tokio::task::spawn(async move {
-            let found_issues = match compute_proposal(model, product_spec, software, progress).await
-            {
-                Ok((new_proposal, system_info, found_issues)) => {
-                    let mut state = state.write().await;
-                    state.proposal.software = Some(new_proposal);
-                    state.system = system_info;
-                    found_issues
-                }
-                Err(error) => {
-                    let new_issue = Issue::new(
-                        "software.proposal_failed",
-                        "It was not possible to create a software proposal",
-                    )
-                    .with_details(&error.to_string());
-                    let mut state = state.write().await;
-                    state.proposal.software = None;
-                    vec![new_issue]
-                }
-            };
-
-            _ = issues.cast(issue::message::Set::new(Scope::Software, found_issues));
-            _ = events.send(Event::ProposalChanged {
-                scope: Scope::Software,
-            });
-        });
+        self.update_proposal().await?;
 
         Ok(())
     }
 }
 
+impl Service {}
```

Comment:
what is the point of this line?

Reply:
It is a test to check whether the reviewer is actually reading the code 😉. It is just a left-over, I will remove it.